### PR TITLE
Tests for App

### PR DESCRIPTION
**Pull request by francistse23**
#81 
- `npm run test-coverage` to run all the tests/tests suites and generate a report locally (add `/test-coverage` to `.gitignore` if you don't want to push this)

- Added tests for the app
- Achieving > 98% statements coverage
- Most components, utils, etc. have been accounted for

### CODE REVIEW

File: src/__tests__/colorConverter.test.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { getColorValue, lightenDarkenColor } from "utils/colorConverter";
+
+import { COLOR_LIBRARY } from "constants/theme";
```
Comment (Kedyn) [inline src/__tests__/colorConverter.test.ts:3]: Never used.

File: src/__tests__/components/Player.test.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import React, { KeyboardEvent } from "react";
```
Comment (Kedyn) [inline src/__tests__/components/Player.test.tsx:1]: KeyboardEvent never used.

File: src/__tests__/components/Sections.test.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+describe("Sections tests", () => {
+  let testStore: MockStore;
+  let dispatchSpy: jest.SpyInstance;
+
+  describe("editing unlocked and not mobile", () => {
+    beforeEach(async () => {
+      const mockStore = configureStore();
+      testStore = mockStore(store.getState());
+      dispatchSpy = jest.spyOn(testStore, "dispatch");
```
Comment (Kedyn) [inline src/__tests__/components/Sections.test.tsx:22]: Assigned but never used.

File: src/utils/i18n.ts [hunk 4]
```diff
@@ -25,6 +25,11 @@ i18n
     backend: {
       loadPath: "./assets/locales/{{lng}}/{{ns}}.json",
     },
+
+    react: {
+      useSuspense: (global as any).__DEV__ ? false : true,
```
Comment (Kedyn) [inline src/utils/i18n.ts:30]: lint gives a warning, but could not find a solution for the global type. Should add a ignore comment:
`// eslint-disable-line` or `// eslint-disable-next-line`.

**Review by Kedyn — APPROVED** [review]
Overall amazing work, just 4 minor linting warnings. Because these are only warnings and do not break anything, I do not think they really need to be addressed. So up to you if you want to address these warnings or just merge. Also, should we be deleting the test files within the components? Once again awesome stuff.

File: src/utils/AUMT/Layer.ts [hunk 5]
```diff
@@ -42,6 +43,14 @@ export default class Layer {
...
   }
 
+  public getEntities(): Array<Entity> {
+    return this.entities;
+  }
+
+  public setEntities(entities: Array<TextLine>): void {
+    this.entities = entities;
+  }
+
```
[the remark this answers is not in the document]
Comment (francistse23) [reply src/utils/AUMT/Layer.ts:53]: yeah was not too familiar with the whole public/private. just needed some way to access the variable to do the assertions

File: src/store/slices/PlayerEditLockSlice.ts [hunk 6]
```diff
@@ -10,16 +10,13 @@ const PlayerEditLockSlice = createSlice({
...
-    resetPlayerEditLockState: () => false,
   },
 });
 
 export const {
   setPlayerEditLock,
 
   togglePlayerEditLock,
-
-  resetPlayerEditLockState,
```
[the remark this answers is not in the document]
Comment (francistse23) [reply src/store/slices/PlayerEditLockSlice.ts:22]: I don't recall making changes to this Slice specifically... let me check the main branch to see what it's supposed to be

[the remark this answers is not in the document]
Comment (francistse23) [reply src/store/slices/PlayerEditLockSlice.ts:22]: hmmm `git pull` didnt get that code for some reason. I added it back in

**Comment by francistse23** [comment]
thanks for pointing out the lint issues. addressed the issues and ran the command again. It should be free of issues now

also removed the unused test files since I grouped them all under `__tests__`

can be merged if you approve the changes

**Review by Kedyn — APPROVED** [review]

**Pull request merged**